examples/DeepState.sol

```diff
@@ -0,0 +1,53 @@
+contract DeepStateProperty {
+
+    event ASSERT_LOG(bytes32 msg);
+
+    function ASSERT(bool x) internal {
+        assert(x);
+    }
+
+    function ASSUME_GT(uint x, uint y) internal {
+        assert(x > y);
+    }
+
+    function ASSUME_GE(int x, int y) internal {
+        assert(x >= y);
+    }
+
+    function ASSUME_LT(uint x, uint y) internal {
+        assert(x < y);
+    }
+
+    function ASSUME_LE(uint x, uint y) internal {
+        assert(x <= y);
+    }
+
+    function ASSUME_NE(uint x, uint y) internal {
+        assert(x != y);
+    }
+
+    function ASSERT_NE(uint x, uint y, bytes32 msg) internal {
+        assert(x != y);
+    }
+
+    function ASSERT_LT(uint x, uint y, bytes32 msg) internal {
+        if (!(x < y)) {
+            ASSERT_LOG(msg);
+            assert(false);
+        }
+    }
+
+    function ASSERT_GT(uint x, uint y, bytes32 msg) internal {
+        if (!(x > y)) {
+            ASSERT_LOG(msg);
+            assert(false);
+        }
+    }
+
+    function BOOL2UINT(bool x) internal returns (uint) {
+        if (x)
+            return 1;
+        else
+            return 0;
+    }
+}
```

Comment:
Can you implement other API methods, e.g. `Min`, `Max`, etc., perhaps hooking these functions by name from the executor, so that you can also implement things like `Pump`?

Reply:
Unfortunately manticore seems to fail to hook functions when it executes a smart contract. I already reported [this issue](https://github.com/trailofbits/manticore/issues/758).